internal/app/machined/pkg/controllers/k8s/control_plane.go

```diff
@@ -121,10 +119,8 @@ func (ctrl *ControlPlaneController) Run(ctx context.Context, r controller.Runtim
 			return fmt.Errorf("error getting machine type: %w", err)
 		}
 
-		machineType := machineTypeRes.(*config.MachineType).MachineType()
-
-		if machineType == machine.TypeWorker {
-			if err = ctrl.teardownAll(ctx, r, logger); err != nil {
+		if !machineType.MachineType().IsControlPlane() {
+			if err = ctrl.teardownAll(ctx, r); err != nil {
```

Comment:
why would we need to delete the resources if it's not controlplane, it shouldn't be there in the first place, right?

Reply:
I tried to explain it above, the previous check got into the state when resource was created for worker nodes while `MachineType` was in `TypeUnknown`. This code is more for completeness, once we support changing `MachineType` on the fly one day.

Reply:
that makes sense. cool